main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -50,6 +52,15 @@ object Unification {
   def unifyVar(x: Type.Var, tpe0: Type, renv: RigidityEnv, lenv: LevelEnv)(implicit flix: Flix): Result[(Substitution, List[Ast.BroadEqualityConstraint]), UnificationError] = {
     // purify the regions of the type that are out of scope
     val tpe = lenv.purify(tpe0)
+
+    // Set the variable levels to the minimum of all flexible variables involved.
+    val tvars = tpe0.typeVars + x
+    val levelOpt = tvars.filter(tv => renv.isFlexible(tv.sym)).map(_.sym.level).minOption
```

Comment:
I wonder about only touching flexible variables.

If x = y! and x is level 3 and y! is level 17, do we not want y! to be marked as escaping to lvl 3, so we can later reject it.

Maybe this is only relevant for illegal programs?

Reply:
In Oleg's examples, QVars (~ rigid vars for us) don't have levels. I need to think about whether this is a problem for escapes.